src/integ_test/java/games/strategy/engine/lobby/server/db/MutedMacControllerTest.java

```diff
@@ -0,0 +1,79 @@
+package games.strategy.engine.lobby.server.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+
+import org.junit.Test;
+
+import games.strategy.util.MD5Crypt;
+import games.strategy.util.Util;
+
+public class MutedMacControllerTest {
+
+  private final MutedMacController controller = spy(new MutedMacController());
+
+  @Test
+  public void testMuteMacForever() {
+    final String username = muteUsername(null);
+    assertTrue(controller.isMacMuted(username));
+    assertEquals(Long.MAX_VALUE, controller.getMacUnmuteTime(username));
+  }
+
+  @Test
+  public void testMuteMac() {
+    final Instant muteUntil = Instant.now().plusSeconds(100L);
+    final String username = muteUsername(muteUntil);
+    assertTrue(controller.isMacMuted(username));
+    assertEquals(muteUntil, Instant.ofEpochMilli(controller.getMacUnmuteTime(username)));
+    when(controller.now()).thenReturn(muteUntil.plusSeconds(1L));
+    assertFalse(controller.isMacMuted(username));
+  }
+
+  @Test
+  public void testUnmuteMac() {
+    final Instant muteUntil = Instant.now().plusSeconds(100L);
+    final String username = muteUsername(muteUntil);
+    assertTrue(controller.isMacMuted(username));
+    assertEquals(muteUntil, Instant.ofEpochMilli(controller.getMacUnmuteTime(username)));
+    controller.addMutedMac(username, Instant.now().minusSeconds(10L));
+    assertFalse(controller.isMacMuted(username));
+  }
+
+  @Test
+  public void testMuteMacInThePast() {
+    final Instant muteUntil = Instant.now().minusSeconds(10L);
+    final String username = muteUsername(muteUntil);
+    assertFalse(controller.isMacMuted(username));
+  }
+
+
+  @Test
+  public void testMuteMacInTooNearFuture() {
+    final Instant muteUntil = Instant.now();
+    when(controller.now()).thenReturn(muteUntil.minusSeconds(10L));
+    final String username = muteUsername(muteUntil);
+    assertFalse(controller.isMacMuted(username));
+  }
+
+  @Test
+  public void testMuteMacUpdate() {
+    final String username = muteUsername(null);
+    assertTrue(controller.isMacMuted(username));
+    assertEquals(Long.MAX_VALUE, controller.getMacUnmuteTime(username));
+    final Instant muteUntill = Instant.now().plusSeconds(100L);
+    controller.addMutedMac(username, muteUntill);
+    assertTrue(controller.isMacMuted(username));
+    assertEquals(muteUntill, Instant.ofEpochMilli(controller.getMacUnmuteTime(username)));
+  }
+
+  private String muteUsername(final Instant length) {
```

Comment:
Problem then would be the uniqueness of the values.
You mentioned this earlier, the test need some sort of rollback

Reply:
In this case, `hashedMac` is an instance field.  Since JUnit creates a new instance of the test class for each test method, the field initializer will run once per test method and generate a unique value per test method.  That should be equivalent to what you had before, right?  Unless there was a case I didn't see where one of the helper methods was called multiple times in a single test method.

Reply:
Of course! 
Should've thought about that, I did rely on that when mocking the controller already ^^